tools/sonarcheck/check.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ ! "$SONAR_TOKEN" ]; then
+  echo "SONAR_TOKEN environment is null, skip check"
+  exit 0
+fi
+./mvnw --batch-mode verify sonar:sonar -Dmaven.test.skip=true -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=apache -Dsonar.projectKey=apache_incubator-seatunnel -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
```

Comment:
Hi @CalvinKirs , this basically says that, the check will only be executed in `push` (not `pull_request`) and those pull requests that are opened in this repo (not fork repo). Meaning the check only check committers' PRs 😂 IS that expected? I think all contributors might bring something that breaks Sonar Check

Reply:
It seems that for security reasons, secrets cannot be passed, here is the explanation of infra, am I wrong? 🤣

Reply:
https://issues.apache.org/jira/browse/INFRA-23101

Reply:
I know that secrets cannot be passed to forked repo. I mean do we have a "local mode" that, we can check and fail the CI, without uploading to Sonar cloud?

Reply:
Looks there is no local mode for Sonar check, let's keep it as is